lib/src/common/polymer_descriptor.dart

```diff
@@ -202,7 +202,13 @@ JsObject _jsType(Type type) {
       return context['String'];
     case 'Map':
     case 'JsObject':
-    default:
       return context['Object'];
+    default:
+      // Store the type in the same manner as done in _addDartInstance so the
+      // type is retrieved in dartValue through _getDartInstance.
+      var value = new JsObject(context['Object']);
```

Comment:
I think its ok to be more lax on the return type, and just return `type` here. We don't really get any value from creating the js object.